packages/gcf-utils/src/gcf-utils.ts

```diff
@@ -151,6 +151,71 @@ export interface CronPayload {
   cron_org: string;
 }
 
+export interface BotSecrets {
+  privateKey: string;
+  appId: string;
+  secret: string;
+}
+
+/**
+ * A helper for fetch secret from SecretManager.
+ */
+async function getBotSecrets(): Promise<BotSecrets> {
+  const projectId = process.env.PROJECT_ID;
+  const functionName = process.env.GCF_SHORT_FUNCTION_NAME;
+  const secretsClient = new SecretManagerV1.SecretManagerServiceClient();
+  const [version] = await secretsClient.accessSecretVersion({
+    name: `projects/${projectId}/secrets/${functionName}/versions/latest`
+  });
+  // Extract the payload as a string.
+  const payload = version?.payload?.data?.toString() || '';
+  if (payload === '') {
+    throw Error('did not retrieve a payload from SecretManager.');
+  }
+  const secrets = JSON.parse(payload);
+
+  if (Object.prototype.hasOwnProperty.call(secrets, 'cert')) {
+    secrets.privateKey = secrets.cert;
+    delete secrets.cert;
+  }
+  if (Object.prototype.hasOwnProperty.call(secrets, 'id')) {
+    secrets.appId = secrets.id;
+    delete secrets.id;
+  }
+  return {
+    privateKey: secrets.privateKey,
+    appId: secrets.appId,
+    secret: secrets.secret,
+  };
+}
+
+/**
+ * A helper for getting an Octokit instance authenticated as an App.
+ */
+export async function getAuthenticatedOctokit(
```

Comment:
Should the Bootstrapper class call this one instead of it's instance method?

Reply:
Not sure

Reply:
I could not make it work without the installation id.

Reply:
Update: I finally make it work without the installation id

Reply:
But it doesn't automatically renew the token upon expiration and the expiration seems to be 10 minutes. It might be long enough for our use case.

I'll add some tests in the canary bot and if it works, I'll change Bootstrapper to use this version.

Reply:
Update: I hope the current version will handle token expiration